src/Symfony/Bundle/FrameworkBundle/Tests/Functional/Bundle/DefaultConfigTestBundle/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Symfony\Bundle\FrameworkBundle\Tests\Functional\Bundle\DefaultConfigTestBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+    public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $treeBuilder = new TreeBuilder('default_config_test');
+
+        $treeBuilder->getRootNode()
+            ->children()
+                ->scalarNode('foo')->defaultValue('bar')->end()
```

Comment:
`defaultValue('%kernel.debug%')` perhaps to see if it behaves the same as if it were configured

Reply:
Added a different default value with a custom parameter. And this actually showed that the `resolveValue` you mentioned was missing and needed.